src/app/add-new-product/add-new-product.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ProductService } from '../services/productService';
+
+@Component({
+  selector: 'app-add-new-product',
+  templateUrl: './add-new-product.component.html',
+  styleUrls: ['./add-new-product.component.css']
+})
+export class AddNewProductComponent implements OnInit {
+
+  myForm: FormGroup;
+
+  constructor(
+    private fb: FormBuilder,
+    private productService: ProductService  
+  ) { }
+
+  ngOnInit() {
+    this.myForm = this.fb.group({
+      id: ['',[
+        Validators.required,
+        Validators.min(1),
+        Validators.pattern('[0-9]*')
+      ]],
+      name: ['',[
+        Validators.required,
+        Validators.minLength(4),
+      ]],
+      category: ['',[
+        Validators.required,
+        Validators.minLength(4)
+      ]],
+      price: ['',[
+        Validators.required,
+        Validators.min(10),
+        Validators.max(5000),
+        Validators.pattern('[0-9]*')
+      ]],
+      description: ['',[
+        Validators.required,
+        Validators.minLength(5)
+      ]]
+    })
+
+    this.myForm.valueChanges.subscribe(console.log);
```

Comment:
**suggestion:** you can remove this line since you use it only for the console.log (also remove any console.log left after debugging)